rpc/internal/test_results_service.go

```diff
@@ -0,0 +1,58 @@
+package internal
+
+import (
+	"context"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+type testResultsService struct {
+	env cedar.Environment
+}
+
+// AttachTestResultsService attaches the test results service to the given gRPC
+// server.
+func AttachTestResultsService(env cedar.Environment, s *grpc.Server) {
+	srv := &testResultsService{
+		env: env,
+	}
+	RegisterCedarTestResultsServer(s, srv)
+}
+
+// CreateTestResultsRecord creates a new test results record in the database.
+func (s *testResultsService) CreateTestResultsRecord(ctx context.Context, info *TestResultsInfo) (*TestResultsResponse, error) {
+	conf := model.NewCedarConfig(s.env)
+	if err := conf.Find(); err != nil {
+		return nil, newRPCError(codes.Internal, errors.Wrap(err, "problem fetching cedar config"))
+	}
+	if conf.Bucket.TestResultsBucketType == "" {
+		conf.Bucket.TestResultsBucketType = model.PailLocal
+	}
+
+	record := model.CreateTestResults(info.Export(), conf.Bucket.TestResultsBucketType)
+	record.Setup(s.env)
+	return &TestResultsResponse{TestResultsRecordId: record.ID},
+		newRPCError(codes.Internal, errors.Wrap(record.SaveNew(ctx), "problem saving test results record"))
+}
+
+// AddTestResults adds test results to an existing test results record.
+func (s *testResultsService) AddTestResults(ctx context.Context, results *TestResults) (*TestResultsResponse, error) {
+	return nil, nil
```

Comment:
can we return errors till tehy're implemented?

Reply:
sure